core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -17,147 +17,150 @@
  * question domain objects.
  */
 
-require('domain/state/StateObjectFactory.ts');
-
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(id, stateData, languageCode, version,
-        linkedSkillIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-    };
-
-    // Instance methods
-
-    Question.prototype.getId = function() {
-      return this._id;
-    };
-
-    Question.prototype.getStateData = function() {
-      return this._stateData;
-    };
-
-    Question.prototype.setStateData = function(newStateData) {
-      this._stateData = angular.copy(newStateData);
-    };
-
-    Question.prototype.getLanguageCode = function() {
-      return this._languageCode;
-    };
-
-    Question.prototype.setLanguageCode = function(languageCode) {
-      this._languageCode = languageCode;
-    };
-
-    Question.prototype.getVersion = function() {
-      return this._version;
-    };
-
-    Question.prototype.getLinkedSkillIds = function() {
-      return this._linkedSkillIds;
-    };
-
-    Question.prototype.setLinkedSkillIds = function(linkedSkillIds) {
-      this._linkedSkillIds = linkedSkillIds;
-    };
-
-    // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    Question['createDefaultQuestion'] = function(skillIds) {
-    /* eslint-enable dot-notation */
-      return new Question(
-        null, StateObjectFactory.createDefaultState(null),
-        DEFAULT_LANGUAGE_CODE, 1, skillIds);
-    };
-
-    Question.prototype.getValidationErrorMessage = function() {
-      var interaction = this._stateData.interaction;
-      if (interaction.id === null) {
-        return 'An interaction must be specified';
-      }
-      if (interaction.hints.length === 0) {
-        return 'At least 1 hint should be specfied';
+// require('domain/state/StateObjectFactory.ts');
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { StateObjectFactory, State } from 'domain/state/StateObjectFactory';
+
+export class Question {
+  id;
```

Comment:
Could we even add types here? (It would copy of what you mentioned in constructor args)

Reply:
Done